api/envoy/api/v2/listener/listener.proto

```diff
@@ -216,4 +227,9 @@ message ListenerFilter {
 
     google.protobuf.Any typed_config = 3;
   }
+
+  // Decide when to enable/disable this listener filter on incoming traffic.
+  oneof switches {
+    ListenerFilterMatch disabled = 4;
```

Comment:
I think calling this `bypass_matcher` would be clearer? Is bypass the behavior you always want?

Reply:
Also, might negative logic be needed in some situations, and we want to enable only on certain ports?

Reply:
my initial intention is `ListenerFilterMatch enabled = 5` so it's `oneof` positive or negative

Reply:
Switch to recursive match message and resolve this